src/defrag.c

```diff
@@ -755,6 +755,15 @@ void defragScanCallback(void *privdata, const dictEntry *de) {
  * or not, a false detection can cause the defragmenter to waste a lot of CPU
  * without the possibility of getting any results. */
 float getAllocatorFragmentation(size_t *out_frag_bytes) {
```

Comment:
Can the linker feature `--wrap` (used via gcc with `-Wl,--wrap`) be leveraged to mock `getAllocatorFragmentation`, providing a less invasive and cleaner solution for testing purposes?

Reply:
nice idea. let me take a look and see if this can be used

Reply:
@xbasel I think that my last change keeps all overriden functions in the allocator_defrag.h so not much value should come from wrapping them and reference by the linker